webcommon/javascript.nodejs/src/org/netbeans/modules/javascript/nodejs/resources/npm-resolver.xml

```diff
@@ -26,4 +26,12 @@
         <name name="package.json"/>
         <resolver mime="text/package+x-json"/>
     </file>
+    <file>
+        <name name="package-lock.json"/>
+        <resolver mime="text/package-lock+x-json"/>
+    </file>
+    <file>
+        <name name="npm-debug.log"/>
+        <resolver mime="text/plain"/>
```

Comment:
This looks fishy. I understand, that you want to associate your new Editor with only "npm-debug.log", yet you assign it the mimeType "text/plain". That mimetype matches a big chunk of files on my harddisk, which you would all pull through your special data object.

Reply:
Oops, I didn't know that. I thought the combination of resolver AND filename in this example is enough to make it unique. So for this I really need a new mimetype, right? text/package+plain or text/npm-debug-log. Looks weirdo. Any other solutions?

Reply:
None I'm aware of - this file just describes how a mimetype is assigned. In your case you are using a name to mime-type mapping.

Reply:
Ok, will change the mimetype, weird but ok. I think it must be possible to add more stuff to a fully qualified unique file name like package.json or in my example to npm-debug.log like adding icons to not change or adding a mimetype. But this is another topic. I will change the mimetype to text/npm-log if this is okish.